plugins/wasm_bpf/wasm-bpf.cpp

```diff
@@ -0,0 +1,258 @@
+  size_t sample_size = size;
+  if (max_poll_size < size) {
+    sample_size = max_poll_size;
+  }
+  memcpy(poll_data, data, sample_size);
+  auto module_inst = wasm_module_instance;
+  WasmEdge_String names[1];
+  /// a valid module instance should have only one table
+  uint32_t exported_table_len =
+      WasmEdge_ModuleInstanceListTable(module_inst, names, std::size(names));
+  assuming(exported_table_len == 1);
+  auto table_inst = WasmEdge_ModuleInstanceFindTable(module_inst, names[0]);
+  assuming(table_inst);
+  WasmEdge_Value value;
+  auto get_data_result =
+      WasmEdge_TableInstanceGetData(table_inst, &value, wasm_sample_function);
+  assuming(WasmEdge_ResultOK(get_data_result));
+  assert(value.Type == WasmEdge_ValType::WasmEdge_ValType_FuncRef);
+  auto func_ref = WasmEdge_ValueGetFuncRef(value);
+
+  WasmEdge_Value invoke_func_params[3] = {
+      WasmEdge_ValueGenI32(wasm_ctx),
+      WasmEdge_ValueGenI32(wasm_buf_ptr),
+      WasmEdge_ValueGenI32(size),
+  };
+  WasmEdge_Value invoke_func_result;
+  auto call_result = WasmEdge_ExecutorInvoke(
+      wasm_executor, func_ref, invoke_func_params, 3, &invoke_func_result, 1);
+  if (!WasmEdge_ResultOK(call_result)) {
+    return -EINVAL;
+  }
+  return WasmEdge_ValueGetI32(invoke_func_result);
+}
+
+/// \brief create a bpf buffer based on the object map type
+std::unique_ptr<bpf_buffer> bpf_buffer__new(struct bpf_map *events) {
+  bpf_map_type map_type = bpf_map__type(events);
+  switch (map_type) {
+  case BPF_MAP_TYPE_PERF_EVENT_ARRAY:
+    return std::make_unique<perf_buffer_wrapper>(events);
+  case BPF_MAP_TYPE_RINGBUF:
+    return std::make_unique<ring_buffer_wrapper>(events);
+  default:
+    return nullptr;
+  }
+}
+
+/// Get the file descriptor of a map by name.
+int32_t wasm_bpf_program::bpf_map_fd_by_name(const char *name) {
+  return bpf_object__find_map_fd_by_name(obj.get(), name);
+}
+/// \brief load all bpf programs and maps in a object file.
+int32_t wasm_bpf_program::load_bpf_object(const void *obj_buf,
+                                          size_t obj_buf_sz) {
+  auto object = bpf_object__open_mem(obj_buf, obj_buf_sz, nullptr);
+  if (!object) {
+    return static_cast<int32_t>(libbpf_get_error(object));
+  }
+  obj.reset(object);
+  return bpf_object__load(object);
+}
+
+/// \brief attach a specific bpf program by name and target.
+int32_t wasm_bpf_program::attach_bpf_program(const char *name,
+                                             const char *attach_target) {
+  struct bpf_link *link;
+  if (!attach_target) {
+    // auto attach base on bpf_program__section_name. The works well for most
+    // bpf types, include kprobe, uprobe, fentry, lsm, etc.
+    link =
+        bpf_program__attach(bpf_object__find_program_by_name(obj.get(), name));
+  } else {
+    struct bpf_object *o = obj.get();
+    struct bpf_program *prog = bpf_object__find_program_by_name(o, name);
+    if (!prog) {
+      spdlog::error("get prog {} fail", name);
```

Comment:
The logging should have a prefix to identify this is generated by your pluginz

Reply:
Fixed now